src/aiodynamo/credentials.py

```diff
@@ -572,6 +572,68 @@ async def fetch_metadata(self, http: HttpImplementation) -> Metadata:
         )
 
 
+@dataclass
+class ProcessCredentialsError(Exception):
+    reason: str
+    return_code: int | None
+    stdout: bytes
+    stderr: bytes
+
+
+@dataclass
+class ProcessCredentials(MetadataCredentials):
+    command: list[str]
+
+    async def fetch_metadata(self, http: HttpImplementation) -> Metadata:
+        process = await asyncio.create_subprocess_exec(
+            *self.command,
+            stdout=asyncio.subprocess.PIPE,
+            stderr=asyncio.subprocess.PIPE,
+            stdin=asyncio.subprocess.PIPE,
+        )
+        stdout, stderr = await process.communicate(b"")
```

Comment:
This would block if the subprocess produces too much data.

Arguably there should be a default timeout too, although, perhaps in this particular library, the caller has a timeout and will cancel `fetch_metadata`.

A simple fix could look smth like:

```py
try:
    stdout, stderr = await asyncio.wait_for(
        process.communicate(b""), timeout=30
    )
except asyncio.TimeoutError:
    process.kill()
    await process.wait()
    raise ProcessCredentialsError("Subprocess timed out", None, b"", b"")
```

I dunno if it's worth it in practice.